src/account/mod.rs

```diff
@@ -11,14 +11,32 @@ use crate::{
 use chrono::prelude::{DateTime, Utc};
 use getset::{Getters, Setters};
 use iota::message::prelude::MessageId;
+use once_cell::sync::OnceCell;
 use serde::{Deserialize, Serialize};
 
-use std::{convert::TryInto, path::PathBuf};
+use std::{
+    collections::HashMap,
+    convert::TryInto,
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
 
 mod sync;
 pub(crate) use sync::{repost_message, RepostAction};
 pub use sync::{AccountSynchronizer, SyncedAccount, TransferMetadata};
 
+type AddressesLock = Arc<Mutex<Vec<IotaAddress>>>;
+type AccountAddressesLock = Arc<Mutex<HashMap<AccountIdentifier, AddressesLock>>>;
```

Comment:
sorry, if this is a dumb question, but I am not very much familiar with `wallet.rs` yet. The typedef just looks a bit weird to me.....but are we sure, that `AddressesLock` (the value stored in the `HashMap`) needs to be protected as well as the `HashMap` it is stored in? That would imply that this `Vec<IotaAddress>` needs to be manipulated somewhere else in the codebase. I wonder, though, why that needs to the case. Couldn't we make those manipulation through the `AddressesLock`.

Reply:
I want to lock the transfer process if a input selection is happening. That's because I don't want to have concurrent transfers using the same input addresses; so it needs the wrapper Mutex.

Reply:
hmmm, okay